### PR TITLE
Feature/typed calculator methods

### PR DESCRIPTION
**Pull request by pimjansen**
Typed the `Calculator` methods. There are 2 methods on which the type is unknown. Anyone with an idea since the first one marked as a string was actually not according the tests

### CODE REVIEW

File: src/Faker/Calculator/Iban.php [hunk 1]
```diff
@@ -25,11 +25,10 @@ public static function checksum($iban)
     }
 
     /**
-     * @param string $match
-     *
+     * @param mixed $match
      * @return int
      */
-    private static function alphaToNumberCallback($match)
+    private static function alphaToNumberCallback($match): int
```
Comment (pimjansen) [inline src/Faker/Calculator/Iban.php:31]: According to the old php spec this should be string but according the tests its an array. Anyone who can confirm?

File: src/Faker/Calculator/Ean.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Ean
      * @param string $digits
      * @return int
      */
-    public static function checksum($digits)
+    public static function checksum(string $digits): int
```
Comment (Nyholm) [inline src/Faker/Calculator/Ean.php:23]: One can consider this a BC break. 

The class is not final and if someone extended this class and override this function, then they will get incompatible types. 

https://3v4l.org/jTe3N

**Review by Nyholm — COMMENTED** [review]
Thank you for this PR. Just a thouht

File: src/Faker/Calculator/Ean.php (hunk 2 above)
Comment (pimjansen) [reply src/Faker/Calculator/Ean.php:23]: Did not think about that to be honest. Only other problem it actual means that everything we do on the lib is almost breaking. What are your thoughts on that?

Comment (Nyholm) [reply src/Faker/Calculator/Ean.php:23]: Yeah. Keeping BC could be tricky of this library. 

I think we should do 2 things. 
1. Mark things with deprecation errors everywhere we **need to**. Using [this package](https://github.com/symfony/deprecation-contracts) is helpful. 
2. Add a BC checker so we dont accidentally do something strange.

**Comment by GrahamCampbell** [comment]
If we are following semver, this needs to go into 1.11, and we'll need to make a new branch for that.

**Comment by sanmai** [comment]
[Composer-semver-wise](https://getcomposer.org/doc/articles/versions.md#caret-version-range-) 1.11 and 1.10 is the same major version. If anything, this has to be 2.0.

**Comment by pimjansen** [comment]
> [Composer-semver-wise](https://getcomposer.org/doc/articles/versions.md#caret-version-range-) 1.11 and 1.10 is the same major version. If anything, this has to be 2.0.

Indeed, i will drop this PR for now since for 2.0 will have a much wider discussion

**Pull request closed without merging**